runtime/src/consensus/tendermint/verifier.rs

```diff
@@ -307,6 +309,71 @@ impl Verifier {
             )));
         }
 
+        // Verify our own RAK is published in registry once per epoch.
+        // This ensures consensus state is recent enough.
+        if cache.last_verified_epoch != epoch {
+            if let Some(rak) = self.protocol.get_public_rak() {
+                let registry_state = RegistryState::new(&state);
+
+                match cache.node_id {
+                    // Node ID is cached, query the node and check for matching RAK.
+                    Some(node_id) => {
+                        let node = registry_state
+                            .node(Context::background(), &node_id)
+                            .map_err(|err| {
+                                Error::VerificationFailed(anyhow!(
+                                    "failed to retrieve node from the registry: {}",
+                                    err
+                                ))
+                            })?;
+                        let node = node.ok_or_else(|| {
+                            Error::VerificationFailed(anyhow!(
+                                "own node ID: '{}' not found in registry state",
+                                node_id,
+                            ))
+                        })?;
+                        if !node.has_rak(&rak, &runtime_header.namespace) {
+                            return Err(Error::VerificationFailed(anyhow!(
+                                "own RAK: '{}' not found in registry state",
+                                rak
+                            )));
+                        }
+                    }
+                    // Node ID not cached, need to scan all registry nodes.
+                    None => {
+                        let nodes = registry_state.nodes(Context::background()).map_err(|err| {
+                            Error::VerificationFailed(anyhow!(
+                                "failed to retrieve nodes from the registry: {}",
+                                err
+                            ))
+                        })?;
+                        let mut found_node: Option<PublicKey> = None;
+                        'outer: for node in nodes {
+                            if let Some(rts) = &node.runtimes {
+                                for rt in rts {
```

Comment:
This is not checking the namespace (it should probably just use `node.has_rak` to ensure consistency).

Reply:
Oh right, I added `has_rak` to avoid duplication here, but then forgot to update this part of code :grin: